bundles/org.openhab.binding.siemensrds/src/main/java/org/openhab/binding/siemensrds/internal/RdsDataPoints.java

```diff
@@ -0,0 +1,685 @@
+            return false;
+        }
+
+        try {
+            Gson gson = new GsonBuilder().registerTypeAdapter(BasePoint.class, new PointDeserializer()).create();
+            nuPoints = gson.fromJson(json, RdsDataPoints.class);
+        } catch (Exception e) {
+            LOGGER.error("refreshUsedValues: exception={}", e.getMessage());
+            return false;
+        }
+
+        synchronized (this) {
+            for (Map.Entry<String, BasePoint> entry : nuPoints.points.entrySet()) {
+                BasePoint nuPoint = entry.getValue();
+                BasePoint exPoint = points.get(entry.getKey());
+
+                if (nuPoint instanceof BooleanPoint && exPoint instanceof BooleanPoint) {
+                    ((BooleanPoint) exPoint).value = ((BooleanPoint) nuPoint).value;
+                } else
+
+                if (nuPoint instanceof TextPoint && exPoint instanceof TextPoint) {
+                    ((TextPoint) exPoint).value = ((TextPoint) nuPoint).value;
+                } else
+
+                if (nuPoint instanceof NumericPoint && exPoint instanceof NumericPoint) {
+                    ((NumericPoint) exPoint).value = ((NumericPoint) nuPoint).value;
+                } else
+
+                if (nuPoint instanceof InnerValuePoint && exPoint instanceof InnerValuePoint) {
+                    ((InnerValuePoint) exPoint).inner.value = ((InnerValuePoint) nuPoint).inner.value;
+                    ((InnerValuePoint) exPoint).inner.presentPriority = ((InnerValuePoint) nuPoint).inner.presentPriority;
+                }
+            }
+        }
+        return true;
+    }
+
+}
+
+/**
+ * private class: a generic data point
+ *
+ * @author Andrew Fiddian-Green - Initial contribution
+ * 
+ */
+class BasePoint {
+    @SerializedName("rep")
+    protected int rep;
+    @SerializedName("type")
+    protected int type;
+    @SerializedName("write")
+    protected boolean write;
+    @SerializedName("descr")
+    protected String descr;
+    @SerializedName("limits")
+    protected float[] limits;
+    @SerializedName("descriptionName")
+    protected String descriptionName;
+    @SerializedName("objectName")
+    protected String objectName;
+    @SerializedName("memberName")
+    protected String memberName;
+    @SerializedName("hierarchyName")
+    protected String hierarchyName;
+    @SerializedName("translated")
+    protected boolean translated;
+
+    private String[] enumVals;
+    private boolean enumParsed = false;
+    protected boolean isEnum = false;
+
+    /*
+     * initialize the enum value list
+     */
+    private boolean initEnum() {
+        if (!enumParsed) {
+            if (descr != null && descr.contains("*")) {
+                enumVals = descr.split("\\*");
+                isEnum = true;
+            }
+        }
+        enumParsed = true;
+        return isEnum;
+    }
+
+    public int getPresentPriority() {
+        return 0;
+    }
+
+    /*
+     * => MUST be overridden
```

Comment:
If `asInt` must be overridden than better make this class `abstract` and the method `asInt` also abstract.

Reply:
Ok, done